fmcapi/api_objects.py

```diff
@@ -4586,17 +4588,25 @@ def parse_kwargs(self, **kwargs):
         if 'applications' in kwargs:
             self.applications = kwargs['applications']
 
-    def acp(self, name):
+    def acp(self, name='', id=''):
+        # either name or id of the ACP should be given
         logging.debug("In acp() for ACPRule class.")
-        acp1 = AccessControlPolicy(fmc=self.fmc)
-        acp1.get(name=name)
-        if 'id' in acp1.__dict__:
-            self.acp_id = acp1.id
+        if id != '':
+            self.acp_id = id
```

Comment:
if user gives ACP ID, then ACP name is not found out automatically, as is the case when the user gives ACP name.
This name resolution will have to be done by doing a get() on the ACPRule object in case this PR is approved.